internal/proxy/util.go

```diff
@@ -609,6 +610,89 @@ func validateSchema(coll *schemapb.CollectionSchema) error {
 	return nil
 }
 
+func validateAndFillFunction(coll *schemapb.CollectionSchema) error {
+	nameMap := lo.SliceToMap(coll.GetFields(), func(field *schemapb.FieldSchema) (string, *schemapb.FieldSchema) {
+		return field.GetName(), field
+	})
+	// validate function
+	for _, function := range coll.GetFunctions() {
+		inputFields := []*schemapb.FieldSchema{}
+		for _, name := range function.GetInputFieldNames() {
+			inputField, ok := nameMap[name]
+			if !ok {
+				return fmt.Errorf("function input field not found %s", function.InputFieldNames)
+			}
+			inputFields = append(inputFields, inputField)
+		}
+
+		err := checkFunctionInputField(function, coll, inputFields)
+		if err != nil {
+			return err
+		}
+
+		outputFields := make([]*schemapb.FieldSchema, len(function.GetInputFieldNames()))
+		lackFields := []int{}
+		for i, name := range function.GetOutputFieldNames() {
+			outputField, ok := nameMap[name]
+			if !ok {
```

Comment:
I think we disallow explicitly declare output fields in schema? Should we return conflict fields error if `ok`? CMIIW

OK:

```python
schema = milvus_client.create_schema()
schema.add_field("id", DataType.INT64, is_primary=True, auto_id=True)
schema.add_field("text", DataType.VARCHAR, max_length=1000)

bm25_function = Function(
    name="text_bm25_emb",
    inputs="text",
    outputs="sparse",
    function_type=FunctionType.BM25,
    params={"k1": 1.2, "b": 0.75},
)

# or:
bm25_function = Function(
    name="text_bm25_emb",
    inputs="text",
    outputs=FieldSchema("sparse", DataType.SPARSE_FLOAT_VECTOR),
    function_type=FunctionType.BM25,
    params={"k1": 1.2, "b": 0.75},
)
```

Not OK:

```python
schema = milvus_client.create_schema()
schema.add_field("id", DataType.INT64, is_primary=True, auto_id=True)
schema.add_field("text", DataType.VARCHAR, max_length=1000)
schema.add_field("sparse", DataType.SPARSE_FLOAT_VECTOR, max_length=1000)

bm25_function = Function(
    name="text_bm25_emb",
    inputs="text",
    outputs="sparse",
    function_type=FunctionType.BM25,
    params={"k1": 1.2, "b": 0.75},
)
```

Reply:
Server don’t care about the exist field was create by  user or sdk. 
Not OK Scene should refuse by SDK.

There will be ok when user use function like
```
schema = milvus_client.create_schema() 
schema.add_field("id", DataType.INT64, is_primary=True, auto_id=True) 
schema.add_field("text", DataType.VARCHAR, max_length=1000)

bm25_function = Function(
   name="text_bm25_emb", 
   inputs="text", 
   FieldSchema("sparse", DataType.SPARSE_FLOAT_VECTOR),
   function_type=FunctionType.BM25, 
   params={"k1": 1.2, "b": 0.75}, 
)
```

And SDK should remove FieldSchema of "sparse" to collection schemas and make sure it not exist.

Reply:
Milvus server should not trust anything passed from SDK, it must perform thorough checks.